cdap-watchdog/src/main/java/co/cask/cdap/logging/gateway/handlers/LogHandler.java

```diff
@@ -90,18 +91,19 @@ public void getRunIdLogs(HttpRequest request, HttpResponder responder, @PathPara
                            @QueryParam("start") @DefaultValue("-1") long fromTimeSecsParam,
                            @QueryParam("stop") @DefaultValue("-1") long toTimeSecsParam,
                            @QueryParam("escape") @DefaultValue("true") boolean escape,
-                           @QueryParam("filter") @DefaultValue("") String filterStr) {
+                           @QueryParam("filter") @DefaultValue("") String filterStr,
+                           @QueryParam("format") @DefaultValue("text") String format) {
     ProgramType type = ProgramType.valueOfCategoryName(programType);
     RunRecordMeta runRecord = programStore.getRun(Id.Program.from(namespaceId, appId, type, programId), runId);
     LoggingContext loggingContext = LoggingContextHelper.getLoggingContextWithRunId(namespaceId, appId, programId, type,
                                                                                     runId, runRecord.getSystemArgs());
 
-    doGetLogs(responder, loggingContext, fromTimeSecsParam, toTimeSecsParam, escape, filterStr, runRecord);
+    doGetLogs(responder, loggingContext, fromTimeSecsParam, toTimeSecsParam, escape, filterStr, runRecord, format);
   }
 
   private void doGetLogs(HttpResponder responder, LoggingContext loggingContext,
                          long fromTimeSecsParam, long toTimeSecsParam, boolean escape, String filterStr,
-                         @Nullable RunRecordMeta runRecord) {
+                         @Nullable RunRecordMeta runRecord, @Nullable String format) {
```

Comment:
I think it is better to have a default value, since we treat "text" as default later too, right?